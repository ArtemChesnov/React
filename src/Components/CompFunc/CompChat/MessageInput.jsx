import React from 'react';

export const MessageInput = (props) => {
  return (
    <>
      <textarea
        type="text"
        className="message-input"
        required
        onChange={props.changeMessage}
        value={props.messageValue}
        placeholder="Напишите сообщение..."
      />
    </>
  );
};
