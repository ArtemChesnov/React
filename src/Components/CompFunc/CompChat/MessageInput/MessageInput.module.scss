.message__input {
    margin: 5px 0 0 0;
    background: rgba(255, 255, 255, 0.15);
    outline: none;
    color: rgba(255, 255, 255, 0.9);
    font-weight: 300;
    border: none;
    padding: 2px 0 2px 10px;
    height: 9vh;
    font-family: BlinkMacSystemFont,
        "Segoe UI",
        "Roboto",
        "Oxygen",
        "Ubuntu",
        "Cantarell",
        "Fira Sans",
        "Droid Sans",
        "Helvetica Neue",
        sans-serif;
    -webkit-font-smoothing: antialiased;

    &::placeholder {
        font-size: 0.9em;
        color: rgba(255, 255, 255, 0.9);
        font-weight: 300;
    }
}