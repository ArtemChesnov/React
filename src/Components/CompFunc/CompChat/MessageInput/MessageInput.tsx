import React, { FC } from 'react';
import style from './MessageInput.module.scss';

interface MessageInputProps {
  messageValue: string;
  setMessageValue: (event: string) => void;
}

export const MessageInput: FC<MessageInputProps> = ({
  messageValue,
  setMessageValue,
}) => {
  return (
    <textarea
      className={style.message__input}
      required
      onChange={(event) => {
        setMessageValue(event.target.value);
      }}
      value={messageValue}
      placeholder="Напишите сообщение..."
    />
  );
};
