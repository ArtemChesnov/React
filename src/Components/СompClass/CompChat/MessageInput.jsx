import React, { Component } from 'react';

export class MessageInput extends Component {
  render() {
    return (
      <>
        <textarea
          type="text"
          className="message-input"
          required
          onChange={this.props.changeMessage}
          value={this.props.messageValue}
          placeholder="Напишите сообщение..."
        />
      </>
    );
  }
}
