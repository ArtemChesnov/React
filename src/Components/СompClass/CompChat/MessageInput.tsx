import React, { Component } from 'react';

interface AutorInputProps {
  messageValue: string;
  messageChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
}

export class MessageInput extends Component<AutorInputProps> {
  render() {
    return (
      <>
        <textarea
          className="message-input"
          required
          onChange={this.props.messageChange}
          value={this.props.messageValue}
          placeholder="Напишите сообщение..."
        />
      </>
    );
  }
}
