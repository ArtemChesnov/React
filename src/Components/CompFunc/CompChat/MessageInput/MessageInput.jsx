import React from 'react';
import './MessageInput.css';

export const MessageInput = ({ messageValue, setMessageValue }) => {
  return (
    <>
      <textarea
        type="text"
        className="message-input"
        required
        onChange={(event) => {
          setMessageValue(event.target.value);
        }}
        value={messageValue}
        placeholder="Напишите сообщение..."
      />
    </>
  );
};
