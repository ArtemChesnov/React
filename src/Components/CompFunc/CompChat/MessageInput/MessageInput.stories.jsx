import React from 'react';
import { MessageInput } from './MessageInput';

export default {
  title: 'Components/MessageInput',
  component: MessageInput,
};

const Template = (args) => <MessageInput {...args} />;

export const Primary = Template.bind({});
