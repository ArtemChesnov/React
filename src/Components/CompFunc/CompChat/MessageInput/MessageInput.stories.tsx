import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { MessageInput } from './MessageInput';

export default {
  title: 'Components/MessageInput',
  component: MessageInput,
} as ComponentMeta<typeof MessageInput>;

const Template: ComponentStory<typeof MessageInput> = (args) => (
  <MessageInput {...args} />
);

export const Primary = Template.bind({});
