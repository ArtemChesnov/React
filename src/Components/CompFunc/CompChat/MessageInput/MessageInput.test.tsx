import React from 'react';
import { MessageInput } from './MessageInput';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('MessageInput', () => {
  it('render component', () => {
    render(<MessageInput messageValue="" setMessageValue={jest.fn()} />);
  });
  it('textarea to be in document', () => {
    render(<MessageInput messageValue="" setMessageValue={jest.fn()} />);
    expect(
      screen.getByPlaceholderText('Напишите сообщение...')
    ).toBeInTheDocument();
  });

  it('render with snapshot', () => {
    const { asFragment } = render(
      <MessageInput messageValue="" setMessageValue={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('multiple render', () => {
    render(
      <>
        <MessageInput messageValue="" setMessageValue={jest.fn()} />
        <MessageInput messageValue="" setMessageValue={jest.fn()} />
      </>
    );
    expect(screen.getAllByRole('textbox').length).toBe(2);
  });

  it('render with value', () => {
    render(<MessageInput messageValue="message" setMessageValue={jest.fn()} />);
    expect(
      screen.getByPlaceholderText('Напишите сообщение...')
    ).toHaveTextContent(/message/);
  });
});
