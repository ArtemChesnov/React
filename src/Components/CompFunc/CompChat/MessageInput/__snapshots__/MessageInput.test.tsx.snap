// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput render with snapshot 1`] = `
<DocumentFragment>
  <textarea
    class="message__input"
    placeholder="Напишите сообщение..."
    required=""
  />
</DocumentFragment>
`;
